pkg/disttask/framework/dispatcher/dispatcher.go

```diff
@@ -544,7 +547,14 @@ func (d *BaseDispatcher) onNextStage() (err error) {
 			taskState := proto.TaskStateRunning
 			if d.Task.Step == proto.StepDone {
 				taskState = proto.TaskStateSucceed
-				logutil.Logger(d.logCtx).Info("all subtasks dispatched and processed, finish the task")
+				logutil.Logger(d.logCtx).Info("task succeed",
+					zap.Stringer("type", d.Task.Type),
+					zap.Int64("task-id", d.Task.ID),
+				)
+				if err = d.OnDone(d.ctx, d, d.Task); err != nil {
+					err = errors.Trace(err)
+					return
+				}
```

Comment:
If onDone failed, we will see "task succeed" multiple times, is it expected?

Reply:
this log seems not that useful, will delete it

Reply:
https://github.com/pingcap/tidb/pull/48928/commits/66f294d9bd670c923775be91654f16c39d32f4f5